pkg/reconciler/taskrun/validate_resources.go

```diff
@@ -88,15 +90,27 @@ func validateParams(paramSpecs []v1beta1.ParamSpec, params []v1beta1.Param) erro
 	if len(missingParamsNoDefaults) > 0 {
 		return fmt.Errorf("missing values for these params which have no default values: %s", missingParamsNoDefaults)
 	}
-	extraParams := list.DiffLeft(providedParams, neededParams)
-	if len(extraParams) != 0 {
-		return fmt.Errorf("didn't need these params but they were provided anyway: %s", extraParams)
+	// If alpha features are enabled, disable checking for extra params.
+	// Extra params are needed to support
+	// https://github.com/tektoncd/community/blob/main/teps/0023-implicit-mapping.md
+	// So that parent params can be passed down to subresources (even if they
+	// are not explicitly used).
+	if config.FromContextOrDefaults(ctx).FeatureFlags.EnableAPIFields != "alpha" {
+		extraParams := list.DiffLeft(providedParams, neededParams)
+		if len(extraParams) != 0 {
+			return fmt.Errorf("didn't need these params but they were provided anyway: %s", extraParams)
+		}
 	}
 
 	// Now that we have checked against missing/extra params, make sure each param's actual type matches
 	// the user-specified type.
 	var wrongTypeParamNames []string
 	for _, param := range params {
+		if _, ok := paramTypes[param.Name]; !ok {
+			// Ignore any missing params - this happens when extra params were
+			// passed to the task that aren't being used.
```

Comment:
Aren't missing params the opposite, aka when a param is defined by a `Task` and not on the `TaskRun` ?

Reply:
Reincluded - I realized after why this is needed.

For cases where we use taskRefs, we can't propagate the implicit param down to the Task to create the 1:1 mapping this validation wants. In these cases, we may have excess params provided by the parameter, a subset of which are actually used / type checked against the resolved task spec - everything else that doesn't match we can just pass through since there's no pairing.